plugins/org.eclipse.elk.alg.layered/src/org/eclipse/elk/alg/layered/graphvisitors/InteractiveLayeredGraphVisitor.java

```diff
@@ -0,0 +1,451 @@
+/*******************************************************************************
+ * Copyright (c) 2020 Kiel University and others.
+ * 
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ * 
+ * SPDX-License-Identifier: EPL-2.0  *******************************************************************************/
+package org.eclipse.elk.alg.layered.graphvisitors;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import org.eclipse.elk.alg.layered.options.CycleBreakingStrategy;
+import org.eclipse.elk.alg.layered.options.LayerConstraint;
+import org.eclipse.elk.alg.layered.options.LayeredOptions;
+import org.eclipse.elk.alg.layered.options.LayeringStrategy;
+import org.eclipse.elk.core.math.KVector;
+import org.eclipse.elk.core.options.CoreOptions;
+import org.eclipse.elk.core.options.Direction;
+import org.eclipse.elk.core.util.IGraphElementVisitor;
+import org.eclipse.elk.graph.ElkConnectableShape;
+import org.eclipse.elk.graph.ElkEdge;
+import org.eclipse.elk.graph.ElkGraphElement;
+import org.eclipse.elk.graph.ElkNode;
+import org.eclipse.elk.graph.ElkPort;
+
+/**
+ * Graph visitor which visits only the root node and recursively steps through the graph
+ * to set interactive option for the layered algorithm.
+ */
+public class InteractiveLayeredGraphVisitor implements IGraphElementVisitor {
+
+    /**
+     * Constant help with node placement, since nodes should not overlap and the their width and height may not be
+     * final.
```

Comment:
I don't understand this description.

Reply:
Node sizes with and without interactive layout might be different. This constant is used as spacing between pseudo positions to make sure that nodes that are not in the same layer do not overlap, since the interactive layout would otherwise put them in the same layer. I will add this to the description.